python/FactorySystem/MooseObject.py

```diff
@@ -9,11 +9,30 @@ def validParams():
 
     return params
 
-  def __init__(self, name, params):
-    self._name = name
-    self._pars = params
-
+  ##
+  # MooseObject can be built via two methods:
+  #   MooseObject(name, params)
+  #   MooseObject(params)
+  def __init__(self, *args, **kwargs):
+
+    if len(args) == 1 and isinstance(args[0], InputParameters):
```

Comment:
This is fine but I'm adding a point to the Python stupidity score. I can't believe this is *the way* to do constructor overloading.